### PR TITLE
Open Kerchunk refs as Virtual Dataset

### PR DESCRIPTION
**Pull request by norlandrhagen**
- [x] Closes https://github.com/TomNicholas/VirtualiZarr/issues/118
- [ ] Tests added
- [ ] User visible changes (including notable bug fixes) are documented in `changelog.md`
- [ ] New functions/methods are listed in `api.rst`

Start of PR to address https://github.com/TomNicholas/VirtualiZarr/issues/118.

Lots of open questions! 
- How should we read `.parquet`files into `KerchunkStoreRefs` to pass into **dataset_from_kerchunk_refs**
- RT'ing json seems to loose `_ARRAY_DIMENSIONS`

Would love some feedback @jsignell!

### CODE REVIEW

**Comment by norlandrhagen** [comment]
Update: @TomNicholas and I dug a bit deeper on the `json` roundtrip failing. On a visual inspection, the underlying structure and data seem identical, but the xarray_testing.assert_equal disagrees.

image
 ...
`vds.lat.data.manifest.dict() == rt_vds.lat.data.manifest.dict()` asserts to True.

**Comment by norlandrhagen** [comment]
Also, some weird behavior where `virtualize.to_kerchunk` seems to be adding `ARRAY_DIMENSIONS`?

image

File: virtualizarr/tests/test_kerchunk.py [hunk 1]
```diff
@@ -232,3 +232,26 @@ def test_FileType():
...
+        "json",
+        pytest.param(
+            "parquet", marks=pytest.mark.xfail(reason="parquet reading not finished")
+        ),
+    ],
+)
+def test_kerchunk_to_virtual_dataset(netcdf4_file, tmpdir, format):
+    vds = open_virtual_dataset(netcdf4_file, indexes={})
+
+    # QUESTION: should these live in a fixture? ex. kerchunk_ref_fpath_json, kerchunk_ref_fpath_parquet
```
Comment (jsignell) [inline virtualizarr/tests/test_kerchunk.py:249]: eh you kind of want the original vds as well as the kerchunk refs so I think it is fine as is.

**Comment by TomNicholas** [comment]
I just tried to merge `main` into this because @kthyng is interesting in picking it up.

> Also, some weird behavior where virtualize.to_kerchunk seems to be adding ARRAY_DIMENSIONS?

I'm pretty sure I fixed this in https://github.com/zarr-developers/VirtualiZarr/pull/153

EDIT: Looks like I broke something in the defaults for the fsspec reader though oops

**Comment by norlandrhagen** [comment]
https://github.com/zarr-developers/VirtualiZarr/pull/251

**Pull request closed without merging**